javalib/src/main/scala/java/net/InetAddress.scala

```diff
@@ -751,7 +734,7 @@ object InetAddress {
 
     val ghnStatus = unistd.gethostname(hostName, MAXHOSTNAMELEN);
     if (ghnStatus != 0) {
-      throw new UnknownHostException(unknownHostExceptionMsg(""))
+      throw new UnknownHostException(fromCString(strerror(errno)))
```

Comment:
I actually don't know what message this has on the JVM. But `: Name or service not known` is probably not it :)

Reply:
Is there an easy way to create a test, then we could try and match if possible?

Reply:
Good question. According to the javadoc `getLocalHost()` 

```
Throws:
UnknownHostException - if the local host name could not be resolved into an address.
```

https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getLocalHost--

But I now no idea how to create that situation artificially.

Reply:
Yes, that seems hard.

Reply:
This is a "should never happen" situation.  If anybody ever gets there, they have far more to be 
concerned about than the text of an error message.  I advise we not invest too much time, emotion,
or energy here.

Arman, are you experiencing this error, or is your concern from inspection?  I ask because I
want to rule out sneak paths that I had not considered. Networking is complex and often the facts
in the world do not comply.

I concur that the ": mumble" could be improved.  The "mumble" portion is one of the many
`gai_strerror()` possible results.  I did it in one place and (should have) called from two
so that both used the same right-hand-side.  The ":" portion indicates a null or empty string
host name.  Java has no good convention for that.  
One could use "<null>:" or  <quote>"": mumble<end_quote>.  As long as one is being inconsistent
one can drop the ":" all together.

Reply:
The message, as is, is a close approximation of what `getaddrinfo()` would return (well, gai_strerror() after
a `getaddrinfo` returning non 0.  It is a summary, which says basically the same thing as "UnknownHostException".
It adds little info. The situation at that point is so broken that additional info is unlikely to lead to relief.

The gai message sits loosely on top of a series of errno messages, each of which is likely to be even more obscure.

I am not against a change to use strerror, just do not see it as helpful to end user or worth the crafting.

Reply:
Windows is reported to have a strerror() in <string.h>. I find that Windows includes sometimes are straight forward
"#include <string.h>" and sometimes need special "#ifdef WIN32".  That is by way of saying that may be a half hour
edit, but I doubt it.   I think that a clib <string.h> would need to be created. I did not see one there. 

If you would like that line to be "strerror", by way of apology for the "missing hostname" mistake, I could
do a follow on PR which made and tested those changes.  I see the value not in this file, but in 
possible future user of a clib <string.h>.

My recommended fix here is to just drop the ": ".

Reply:
re: creating the conditions where this message is thrown.

To evoke this message manually, I believe that one must: 

1) Ensure that the kernel is compiled without a hostname.

2) Ensure that '/etc/hostname' (on Ubuntu, names may vary) either does not exist
    or is empty.

3) Ensure that either the IP network is not started after system reboot or
    that it is started without a  bind (name) server.  This is harder than it
    sounds, so not starting the network at all is more reliable.

4) execute a program which calls `getLocalHost()`.

Since the SN test framework uses the network to communicate
with the generated test program, one must figure out the "no name server"
part. 

On POSIX systems, a motivated person can avoid `gethostname()` and 
try to bypass the nameservers by fussing with `uname` (machine name).
Doing this on Windows is left as an exercise for the reader (hence, the use
of `gethostname()` in this code.

One can get into the interesting situation where the kernel host name differs
from the IP (bind) hostname and tries to use the former for the latter.

Reply:
> Arman, are you experiencing this error, or is your concern from inspection? 

Thanks. I am not experiencing this error, it was just a concern from inspection. Happy to replace this with whatever you suggest :)